lib/phoenix_pubsub/local.ex

```diff
@@ -0,0 +1,230 @@
+defmodule Phoenix.PubSub.Local do
+  @moduledoc """
+  PubSub implementation for handling local-node process groups.
+
+  This module is used by Phoenix pubsub adapters to handle
+  their local node subscriptions and it is usually not accessed
+  directly. See `Phoenix.PubSub.PG2` for an example integration.
+  """
+
+  use GenServer
+
+  @doc """
+  Starts the server.
+
+    * `server_name` - The name to register the server under
+
+  """
+  def start_link(server_name, gc_name) do
+    GenServer.start_link(__MODULE__, {server_name, gc_name}, name: server_name)
+  end
+
+  @doc """
+  Subscribes the pid to the topic.
+
+    * `pubsub_server` - The registered server name or pid
+    * `pid` - The subscriber pid
+    * `topic` - The string topic, for example "users:123"
+    * `opts` - The optional list of options. Supported options
+      only include `:link` to link the subscriber to local
+
+  ## Examples
+
+      iex> subscribe(:pubsub_server, self, "foo")
+      :ok
+
+  """
+  def subscribe(pubsub_server, pool_size, pid, topic, opts \\ []) when is_atom(pubsub_server) do
+    {:ok, {topics, pids}} =
+      pubsub_server
+      |> local_for_pid(pid, pool_size)
+      |> GenServer.call({:subscribe, pid, topic, opts})
+
+    true = :ets.insert(topics, {topic, {pid, opts[:fastlane]}})
+    true = :ets.insert(pids, {pid, topic})
+
+    :ok
+  end
+
+  @doc """
+  Unsubscribes the pid from the topic.
+
+    * `pubsub_server` - The registered server name or pid
+    * `pid` - The subscriber pid
+    * `topic` - The string topic, for example "users:123"
+
+  ## Examples
+
+      iex> unsubscribe(:pubsub_server, self, "foo")
+      :ok
+
+  """
+  def unsubscribe(pubsub_server, pool_size, pid, topic) when is_atom(pubsub_server) do
+    {local_server, gc_server} =
+      pid
+      |> pid_to_shard(pool_size)
+      |> pools_for_shard(pubsub_server)
+
+    :ok = Phoenix.PubSub.GC.unsubscribe(pid, topic, local_server, gc_server)
+  end
+
+  @doc """
+  Sends a message to all subscribers of a topic.
+
+    * `pubsub_server` - The registered server name or pid
+    * `topic` - The string topic, for example "users:123"
+
+  ## Examples
+
+      iex> broadcast(:pubsub_server, self, "foo")
```

Comment:
Actually, not sorry. All of these docs are outdated from the time we added shards. :)